src/main/kotlin/com/cultureamp/eventsourcing/BatchedAsyncEventProcessor.kt

```diff
@@ -66,6 +67,7 @@ class BatchedAsyncEventProcessor<M : EventMetadata>(
     )
 
     fun processOneBatch(): Action {
+        lockBookmark(bookmarkName)
```

Comment:
If the idea that this is _refreshing_ the lock, rather than a new lock each time, maybe we should call this `refreshLock` or `refreshLongLivedLock`

Reply:
That's the idea _in the implementation that we have_ but this is an interface into that. Other implementations may lock freshly each time. In fact one concern I have is that `pg_advisory_lock()` keeps track of the number of times it's locked (so it will require that many unlocks before releasing) - if that's stored as an `int` it will allow 2-4 billion locks and then will likely cause some error. So perhaps we need to not refresh the lock but just ensure we have it somehow?

Reply:
Interesting - you're deep down in the bowels at the moment, I don't have enough knowledge about the locking mechanisms in postgres to comment 😬

Reply:
> it will allow 2-4 billion locks and then will likely cause some error
That said, if timeout is one minute it will take an uptime of 3805 years to trigger this

Reply:
> That said, if timeout is one minute it will take an uptime of 3805 years to trigger this

oops I'm confused - timeout has nothing to do with this, it's just how many times we poll - which is likely to be much more frequently than that. if we poll on average 10x second it will take 6 yrs to reach 2 billion. Still ok, but that's assuming it's an `int` not a `short` or even a `byte`